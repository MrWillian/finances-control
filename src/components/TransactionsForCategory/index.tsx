import React from 'react';
import { ScrollView, Text } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import CategoryGradient from '../Gradients/CategoryGradient';

import { CategoryBalance } from '../../../core/lib/adapters/redux/store/ducks/balance';

import { 
  Container, 
  Header, 
  LinkToAll, 
  Label,
  TransactionsForCategoryCard, 
  CategoryIcon,
  CategoryName,
  CategoryValue,
} from './styles';

interface Props {
  categories?: CategoryBalance[];
}

const TransactionsForCategory: React.FC<Props> = ({ categories }) => {

  const iconNameResolve = (name: string) => {
    switch(name) {
      case 'Comida':
        return 'fast-food';
      case 'Entretenimento': 
        return 'game-controller';
      default:
        return 'md-list-circle';
    }
  }

  return (
    <Container>
      <Header>
        <Label>Movimentações por Categoria</Label>
        <LinkToAll>
          <Label>Todas</Label>
          {/* <Icon name="ios-funnel" size={10} color="#C8C8C8" /> */}
        </LinkToAll>
      </Header>

      <ScrollView>
        {categories && categories.length > 0 ?
          categories.map((category: CategoryBalance, index) => 
            <TransactionsForCategoryCard key={index}>
              <CategoryGradient>
                <CategoryIcon>
                  <Icon name={iconNameResolve(category.name)} size={20} color="#FFF" />
                </CategoryIcon>
              </CategoryGradient>

              <CategoryName>{category.name}</CategoryName>
              <CategoryValue>{category.name === 'Lucro' ? '+' : '-'} R$ {category.total}</CategoryValue>
            </TransactionsForCategoryCard>  
          )
          : 
          <Text style={{fontFamily: 'Comfortaa-SemiBold', color: '#FFF', marginTop: 15}}>
            Nenhuma movimentação criada...
          </Text>
        }
      </ScrollView>

    </Container>
  );
}

export default TransactionsForCategory;