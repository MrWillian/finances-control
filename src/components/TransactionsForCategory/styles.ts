import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
`;

export const Header = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

export const LinkToAll = styled.TouchableOpacity`
  width: 18%;
  height: 100%;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;

  background-color: #222f52;
  border-radius: 25px;
  padding: 5px;
`;

export const Label = styled.Text`
  font-family: Comfortaa-Medium;
  font-size: 12px;
  color: #C8C8C8;
`;

export const TransactionsForCategoryCard = styled.View`
  flex: 1;
  margin: 10px 0 0 0;

  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

export const CategoryIcon = styled.View``;

export const CategoryName = styled.Text`
  font-family: Comfortaa-Bold;
  font-size: 14px;
  color: #C8C8C8;
`;

export const CategoryValue = styled.Text`
  font-family: Comfortaa-Bold;
  font-size: 12px;
  color: #FFF;
`;
