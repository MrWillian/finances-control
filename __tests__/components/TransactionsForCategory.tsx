import React from 'react';
import renderer from 'react-test-renderer';

import TransactionsForCategory from '../../src/components/TransactionsForCategory';

it('renders correctly with defaults', () => {
  const button = renderer
    .create(<TransactionsForCategory />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
