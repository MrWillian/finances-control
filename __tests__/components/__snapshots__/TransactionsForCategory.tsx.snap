// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#C8C8C8",
            "fontFamily": "Comfortaa-Medium",
            "fontSize": 12,
          },
        ]
      }
    >
      Movimentações por Categoria
    </Text>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#222f52",
          "borderRadius": 25,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "space-around",
          "opacity": 1,
          "paddingBottom": 5,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 5,
          "width": "18%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#C8C8C8",
              "fontFamily": "Comfortaa-Medium",
              "fontSize": 12,
            },
          ]
        }
      >
        Todas
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "color": "#FFF",
            "fontFamily": "Comfortaa-SemiBold",
            "marginTop": 15,
          }
        }
      >
        Nenhuma movimentação criada...
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
